File: src/commands/client_side/ReadCommand.java
package commands.client_side;

import commands.client_side.core.ServerSideCommand;
import commands.client_side.core.ClientCallbackFunction;
import OrganizationDatabase.OrganizationManagerInterface;

public class ReadCommand extends ServerSideCommand {

    public ReadCommand(ClientCallbackFunction clientCallbackFunction, OrganizationManagerInterface organizationDatabase) {
        super(clientCallbackFunction, organizationDatabase);
    }

    @Override
    protected void executeImplementation(String[] args, ClientCallbackFunction callback) {
        String filename = args[0];
        callback.invoke(organizationDatabase.loadFromFile(filename), null, filename);
    }
}

File: src/commands/client_side/ReadCommand.kt
package commands.client_side

import commands.client_side.core.ServerSideCommand
import database.OrganizationManagerInterface
import exceptions.FileReadException
import lib.ExecutionStatus

class ReadCommand(organizationDatabase: OrganizationManagerInterface) :
    ServerSideCommand(organizationDatabase) {
    override suspend fun executeImplementation(argument: Any?): Result<Unit?> {
        val filename = argument as String

        if (organizationDatabase.loadFromFile(filename) == ExecutionStatus.FAILURE) {
            return Result.failure(FileReadException(filename))
        }

        return Result.success(null)
    }
}
